### DISCUSSION TITLE
allureCypress support for cucumber

### DISCUSSION THREAD

**Discussion by mysticdevx** [allure-framework/allure2#2409] [Questions & Support] [answered]
Hi all!
Great thanks for the the allureCypress plugin! Although the autors of the packages available and contributors put effort to maintain, it will always be good to have the plugin from the source. 

We all benefit a lot from the allure report. Its the best reporting tool by far! What we need is to have it support for the cypress with cucumber frameworks. The well maintained cucumber preprocessor can be found [here](https://www.npmjs.com/package/@badeball/cypress-cucumber-preprocessor). At the moment when I use the allureCypress plugin I just get the scenario result, if its passed. No more details about steps, cypress commands etc.

If it will be helpful, I can gladly submit a minimal repo with Cypess  and cucumber.
Thanks!
(fyi @baev)

**Comment by epszaw** [comment]
Hey there! Will add cucumber support as soon as it possible! 🙏

**Reply by achraf-elk** [reply]
I've encountered an issue while running my Cypress tests with the new allure-cypress plugin alongside cypress-cucumber-preprocessor. The error occurs during the before:spec event handler:

`An error was thrown in your plugins file while executing the handler for the before:spec event.
The error we received was:
Error: Unexpected state in beforeSpecHandler: test-finished. This almost always means that you or some other plugin, are overwriting this plugin's event handlers. For more information & workarounds, see https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/event-handlers.md (if neither workaround work, please report at https://github.com/badeball/cypress-cucumber-preprocessor)
`

I understand that the current allure-cypress plugin does not fully support Cucumber yet. Could you confirm if this issue might be resolved once full support for Cucumber is added to the allure-cypress package in the future?

**Reply by ameduza** [reply]
Hi @achraf-elk , I may be wrong, but it seems to me the issue you described is not strictly related to Allure plugin. It is rather Cypress known issue. I was able to work around it with using custom plugin 'cypress-on-fix' from Gleb Bahmutov: https://github.com/bahmutov/cypress-on-fix

here is my cypress.config.ts cut:
```typescript
import { defineConfig } from 'cypress';
import { addCucumberPreprocessorPlugin } from '@badeball/cypress-cucumber-preprocessor';
import createBundler from '@bahmutov/cypress-esbuild-preprocessor';
import createEsbuildPlugin from '@badeball/cypress-cucumber-preprocessor/esbuild';
import { allureCypress } from 'allure-cypress/reporter';
import cypressOnFix from 'cypress-on-fix';

const cypressJsonConfig = {
  viewportWidth: 1920,
  viewportHeight: 1080,
...
  env: {
    filterSpecs: false,
    omitFiltered: false,
    ...
  },
};

async function setupNodeEvents(
  on: Cypress.PluginEvents,
  config: Cypress.PluginConfigOptions,
): Promise<Cypress.PluginConfigOptions> {
  // This is used as a workaround to deal with the before:spec handler conflict with plugins
  const fixedOn = cypressOnFix(on);

  // This is required for the preprocessor to be able to generate JSON reports after each run, and more,
  await addCucumberPreprocessorPlugin(fixedOn, config);

  fixedOn(
    'file:preprocessor',
    createBundler({
      plugins: [createEsbuildPlugin(config)],
    }),
  );

  allureCypress(fixedOn);
  // Make sure to return the config object as it might have been modified by the plugin.
  return config;
}

export default defineConfig({
  e2e: {
    ...cypressJsonConfig,
    setupNodeEvents,
  },
});
```

**Reply by pradeepkumarj-sportsbet** [reply]
> Hey there! Will add cucumber support as soon as it possible! 🙏

Hi, Still not getting the step details in the report while using cucumber preprocessor.

**Comment by samrasaeed15** [comment]
@ameduza , allure report does not includes details steps. It just includes the name of scenario not all steps. Am i missing something?
i am using badeball/cypress-cucumber-preprocessor

**Reply by ameduza** [reply]
@samrasaeed15 you right. at this moment this is not fully supported. So that is the reason of having this issue in a tracker, I mean #2409

**Comment by epszaw** [comment, marked as answer]
The feature was implemented [here](https://github.com/allure-framework/allure-js/pull/1032) and will be available in the next release.

See examples here:

cjs: https://github.com/epszaw/allure-js-3-examples/tree/master/cypress-cucumber-cjs-javascript
esm: https://github.com/epszaw/allure-js-3-examples/tree/master/cypress-cucumber-esm-javascript

**Reply by baev** [reply]
available in allure-cypress@3.0.0-beta.5

**Reply by Pradeepkumar-github** [reply]
Thanks, but when I use this version I'm getting an error while executing the test. But if I revert to the previous allure-cypress version I'm not getting this error .
"_Converting circular structure to JSON
--> starting at object with constructor 'HTMLDivElement'
|     property '__reactFiber$6yf4zlnpxhq' -> object with constructor 'Xc'
--- property 'stateNode' closes the circle_"

**Reply by epszaw** [reply]
Ok, will check the issue!

**Reply by epszaw** [reply]
Please provide an example where I'll be able to reproduce the problem. And I guess, the problem doesn't relate to the original problem, so it would be nice to continue figuring out in a separated discussion or issue. Could you also open one with the requested details? It could make fixing the problem much faster.

**Reply by ameduza** [reply]
Opened a [new issue](https://github.com/allure-framework/allure-js/issues/1059) about this support. Still not fully supported in my opinion. That blocks us to fully switch to that setup

**Reply by baev** [reply]
The fix is available in the [allure-cypress@3.0.7](https://www.npmjs.com/package/allure-cypress/v/3.0.7)

**Comment by DevPerers** [comment]
Hello
My issue was resolved. I have a problem with Allure report integration with the latest Cypress version and Cucumber. For Cucumber I was using "https://www.npmjs.com/package/cypress-cucumber-preprocessor". I removed that and built a project with "https://www.npmjs.com/package/@badeball/cypress-cucumber-preprocessor".
My Cypress version is "13.17.0"
